mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,195 @@
+import mlflow
+from mlflow.models.evaluation.base import (
+    ModelEvaluator,
+    EvaluationMetrics,
+    EvaluationArtifact,
+)
+from mlflow.utils.file_utils import TempDir
+from mlflow.tracking.artifact_utils import get_artifact_uri
+from sklearn import metrics as sk_metrics
+import matplotlib.pyplot as pyplot
+import scikitplot
+import shap
+import math
+
+shap.initjs()
+
+"""
+[P0] Accuracy: Calculates how often predictions equal labels.
+[P0] BinaryCrossentropy: Computes the crossentropy metric between the labels and predictions.
+[P0] Hinge: Computes the hinge metric between y_true and y_pred.
+[P0] Sum: Computes the (weighted) sum of the given values.
+[P0] Mean: Computes the (weighted) mean of the given values.
+[P0] ExampleCount: Computes the total number of evaluation examples.
+[P0] MeanAbsoluteError: Computes the mean absolute error between the labels and predictions.
+[P0] MeanSquaredError: Computes the mean squared error between y_true and y_pred.
+[P0] RootMeanSquaredError: Computes root mean squared error metric between y_true and y_pred.
+
+[P0] TrueNegatives: Calculates the number of true negatives.
+[P0] TruePositives: Calculates the number of true positives.
+[P0] FalseNegatives: Calculates the number of false negatives.
+[P0] FalsePositives: Calculates the number of false positives.
+https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix
+
+[P0] Precision: Computes the precision of the predictions with respect to the labels.
+[P0] Recall: Computes the recall of the predictions with respect to the labels.
+[P0] AUC: Approximates the AUC (Area under the curve) of the ROC or PR curves.
+[P0] F1 Score: 2*precision*recall / (precision+recall)
+
+[P0] BinaryClassConfusionMatrix
+
+Plots
+[P0] Confusion matrix
+[P0] Interactive ROC curve with metrics (TP/TN/FP/FN/Acc/F1/AUC), binary classification
```

Comment:
Offline discussion: clarify what `Interactive ROC curve with metrics` is.

Reply:
Example: https://docs.datarobot.com/en/docs/modeling/analyze-models/evaluate/roc-curve-tab/index.html

Reply:
```python
fpr, tpr, thresholds = roc_curve(y, pred)
optimal_cutoff = thresholds[np.argmax(tpr - fpr)]
```
(just for reference for our discussions - not suggesting an implementation here)